tests/unittests/test_net.py

```diff
@@ -8220,6 +8220,19 @@ def test_ib(self):
         }
         self.assertEqual(expected, result)
 
+    def test_duplicate_ignored_macs(self):
+        # LP: #199792
+        self._data = copy.deepcopy(self._data)
+        self._data["macs"]["swp0"] = "9a:57:7d:78:47:c0"
+        self._data["macs"]["swp1"] = "9a:57:7d:78:47:c0"
+        self._data["own_macs"].append("swp0")
+        self._data["own_macs"].append("swp1")
+        self._data["drivers"]["swp0"] = "mscc_felix"
+        self._data["drivers"]["swp1"] = "mscc_felix"
+        self._mock_setup()
+        net.get_interfaces_by_mac()
```

Comment:
We could use  does_not_raise and could check logs
```suggestion
        with does_not_raise():
            net.get_interfaces_by_mac()
        msg = (
            "both 'swp1' and 'swp0' have mac '9a:57:7d:78:47:c0'."
            " Ignoring 'swp1' due to driver 'mscc_felix' and 'swp0' having"
            " driver 'mscc_felix'."
        )
        self.assertIn(msg, self.logs.getvalue())
```